app/models/miq_provision_virt_workflow.rb

```diff
@@ -218,7 +218,17 @@ def available_vlans_and_hosts(options = {})
       # TODO: Use Active Record to preload this data?
       MiqPreloader.preload(hosts, :switches => :lans)
       hosts.each do |h|
-        h.lans.each { |l| vlans[l.name] = l.name if @vlan_options[:dvs] || !l.switch.shared? }
+        h.lans.each do |l|
+          next if l.switch.shared? && !@vlan_options[:dvs]
+
+          key = if l.switch.shared?
```

Comment:
@agrare The logic here looks correct, but I am wondering if it should be moved to the [ManageIQ/manageiq-providers-vmware](https://github.com/ManageIQ/manageiq-providers-vmware) repo?

Is there any reason to keep this as base provision_workflow logic?  No other providers need this that I am aware of.